### PR TITLE
mgr/cephadm: update osd removal report immediately

### PR DESCRIPTION
**Pull request by bk201**
Currently, the removal report is updated after entering the main serve()
loop. This tiny window makes clients (like Dashboard) fail to poll the
result. Refresh the report data immediately after scheduling OSD for
removal.

The report structure is changed from Dict to Set because:
- The key is `OSDRemoval` instance, which make it hard to use.
- More consistent with orchestrator interfaces: most calls return a list.

Signed-off-by: Kiefer Chang <kiefer.chang@suse.com>

Dependency: https://github.com/ceph/ceph/pull/33602
The PR contains un-merged commits in #33602, change outside dependent PR is in https://github.com/ceph/ceph/commit/87decc1c6e1bd5538cc6d871e6e2add638ebf84c

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/mgr/cephadm/osd.py [hunk 1]
```diff
@@ -3,21 +3,30 @@
...
-from typing import List, NamedTuple, Dict, Any, Set, Union
+from typing import List, Dict, Any, Set, Union
 
 import orchestrator
 from orchestrator import OrchestratorError
 
 logger = logging.getLogger(__name__)
 
 
-class OSDRemoval(NamedTuple):
```
Comment (sebastian-philipp) [inline src/pybind/mgr/cephadm/osd.py]: `NamedTuple` was problematic?

Comment (bk201) [reply src/pybind/mgr/cephadm/osd.py]: It doesn't allow in-place updating (Unless using some [hacks](https://stackoverflow.com/a/31253184)). I update `pg_count` later. 

Seems CentOS7 ships Python 3.6. If we have at least 3.7, then [dataclasses](https://docs.python.org/3.8/library/dataclasses.html) make life much easier in this case.

Comment (sebastian-philipp) [reply src/pybind/mgr/cephadm/osd.py]: cc @jschmid1 ^^

Comment (jschmid1) [reply src/pybind/mgr/cephadm/osd.py]: well.. I don't think we have 3.7 available across the board. I'd also like to have dataclasses but I think that's not possible _right_now_

**Pull request merged**